### DISCUSSION TITLE
Does this library work with React?

### DISCUSSION THREAD

**Discussion by Sniper3945** [swup/swup#787] [Help] [answered]
Does this library work with react js ?

**Comment by daun** [comment, marked as answer]
It depends 🪴 Swup is framework-agnostic in that doesn't make any assumptions about how your site is rendered. It just expects the server to return valid full-page html for each request. So, if you're using React for server-side rendering, then yes, swup is a great option! If you're using React for client-side routing, the two libraries would just get in each other's way and there'd be better options out there tightly coupled to react-router or similar.
